lldb/include/lldb/DataFormatters/TypeSummary.h

```diff
@@ -258,6 +258,9 @@ class TypeSummaryImpl {
 
   virtual std::string GetDescription() = 0;
 
+  virtual ConstString GetName() = 0;
+  virtual ConstString GetImplType() = 0;
+
```

Comment:
We should switch to `std::string` instead of `ConstString.. These objects that get returned here don't start out as `ConstString` objects. We only want to use `ConstString` objects for things we are comparing many things by name where there will be many instances of objects that have this name.

Reply:
Phew,

I 100% agree, this was getting quite messy